scio/src/main/scala/com/spotify/featran/scio/TFExampleScollectionFunctions.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.scio
+
+import com.spotify.featran.{FeatureExtractor, MultiFeatureExtractor}
+import com.spotify.scio.ScioContext
+import com.spotify.scio.io.Tap
+import com.spotify.scio.tensorflow._
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.io.Compression
+import org.tensorflow.{example => tf}
+
+import scala.concurrent.Future
+
+
+private object FeatranTFRecordSpec {
+
+  def fromFeatureSpec(featureNames: SCollection[Seq[String]]): TFRecordSpec = {
+    TFRecordSpec.fromSColSeqFeatureInfo(
+      featureNames.map(_.map(n => FeatureInfo(n, FeatureKind.FloatList, Map()))))
+  }
+
+  def fromMultiSpec(featureNames: SCollection[Seq[Seq[String]]]): TFRecordSpec = {
+    val featureInfos = featureNames.map(_.zipWithIndex.flatMap {
+      case (sss, i) => sss.map(n =>
+        FeatureInfo(n, FeatureKind.FloatList, Map("multispec-id" -> i.toString)))
```

Comment:
Just want to make sure if I understand this, so this `Map` might as well be a `Option[String]` right?

Reply:
Right, the way this is right now, there is either 0 entries or 1 entry in the Map, so it did not have to be a Map.
It's a bit of speculative engineering, but I thought it would be convenient to have a way to tag feature with whatever we want. This provides much more flexibility to TFRecordSpec.